src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -1127,14 +1128,19 @@ internal EntityHandle TryGetTypeHandle(Type type)
                 return eb._typeBuilder._handle;
             }
 
-            if (IsConstructedFromTypeBuilder(type))
+            if (IsConstructedFromTypeBuilder(type) ||
+                HasElementTypeOfTypeBuilder(type))
```

Comment:
Should this check be in the `IsConstructedFromTypeBuilder` so that it is applied to generic argument too? For example, consider a type like: `Dictionary<MyRefEmittedType[]. int>`.

Reply:
Yes, it should, added and tested, thank you!